runtime/ignite/iginite.go

```diff
@@ -0,0 +1,402 @@
+	})
+
+	vm.Spec.CopyFiles = copyFiles
+
+	// Setting up env variables
+	fcReqKey := igniteConstants.IGNITE_SANDBOX_ENV_VAR + "FIRECRACKER_GO_SDK_REQUEST_TIMEOUT_MILLISECONDS"
+	fcInitKey := igniteConstants.IGNITE_SANDBOX_ENV_VAR + "FIRECRACKER_GO_SDK_INIT_TIMEOUT_SECONDS"
+	vm.SetAnnotation(fcReqKey, "1000")
+	vm.SetAnnotation(fcInitKey, "1")
+
+	// Instructing ignite to connect extra interfaces as tc redirect
+	for _, extraIntf := range extraIntfs {
+		vm.SetAnnotation(igniteConstants.IGNITE_INTERFACE_ANNOTATION+extraIntf, "tc-redirect")
+	}
+
+	providers.Client.VMs().Set(vm)
+
+	err = dmlegacy.AllocateAndPopulateOverlay(vm)
+	if err != nil {
+		log.Printf("Error AllocateAndPopulateOverlay: %s", err)
+		return nil, err
+	}
+
+	vmChans, err := operations.StartVMNonBlocking(vm, c.config.Debug)
+	if err != nil {
+		return nil, err
+	}
+
+	node.NSPath, err = c.GetNSPath(ctx, vm.PrefixedID())
+	if err != nil {
+		return nil, err
+	}
+	return vmChans, utils.LinkContainerNS(node.NSPath, node.LongName)
+}
+
+func (c *IgniteRuntime) StartContainer(ctx context.Context, _ string) error {
+	// this is a no-op
+	return nil
+}
+func (c *IgniteRuntime) StopContainer(ctx context.Context, name string, timeout *time.Duration) error {
+	// this is a no-op, only used by ceos at this stage
+	return nil
+}
+
+func (c *IgniteRuntime) ListContainers(ctx context.Context, gfilters []*types.GenericFilter) ([]types.GenericContainer, error) {
+
+	var result []types.GenericContainer
+
+	var labelStrings []string
+	for _, gf := range gfilters {
+		if gf.FilterType == "label" && gf.Operator == "=" {
+			labelStrings = append(labelStrings, fmt.Sprintf("{{.ObjectMeta.Labels.%s}}=%s", gf.Field, gf.Match))
+		}
+	}
+
+	allVMs, err := providers.Client.VMs().FindAll(filter.NewVMFilterAll("", true))
+	if err != nil {
+		return result, fmt.Errorf("failed to list all VMs: %s", err)
+	}
+
+	if len(labelStrings) < 1 {
+		return c.produceGenericContainerList(allVMs)
+	}
+
+	metaFilter := strings.Join(labelStrings, ",")
+	filters, err := filter.GenerateMultipleMetadataFiltering(metaFilter)
+	if err != nil {
+		return result, fmt.Errorf("failed to generate filters: %s", err)
+	}
+
+	filteredVMs := []*api.VM{}
+	for _, vm := range allVMs {
+		isExpected, err := filters.AreExpected(vm)
+		if err != nil {
+			continue
+		}
+		if isExpected {
+			filteredVMs = append(filteredVMs, vm)
+		}
+	}
+
+	return c.produceGenericContainerList(filteredVMs)
+}
+
+func (c *IgniteRuntime) GetContainer(ctx context.Context, containerID string) (*types.GenericContainer, error) {
+	var result *types.GenericContainer
+	vm, err := providers.Client.VMs().Find(filter.NewVMFilter(containerID))
+	if err != nil {
+		return result, err
+	}
+
+	genericCtrs, err := c.produceGenericContainerList([]*api.VM{vm})
+	if err != nil {
+		return result, err
+	}
+	if len(genericCtrs) != 1 {
+		return result, fmt.Errorf("unexpected number of matched containers %d", len(genericCtrs))
+	}
+
+	return &genericCtrs[0], nil
+}
```

Comment:
the GetContainer doesn't seem to be part of the Runtime interface
Why was it added?

Reply:
I added it initially as a way to ListContainers and probably forgot to remove it.